portal-impl/test/integration/com/liferay/portlet/service/persistence/GroupFinderTest.java

```diff
@@ -170,6 +181,44 @@ public void testFindByCompanyId() throws Exception {
 	}
 
 	@Test
+	public void testFindByCompanyIdByUserGroupGroup() throws Exception {
+		_userGroup = UserGroupTestUtil.addUserGroup();
+		_userGroupGroup = GroupTestUtil.addGroup();
+		_userGroupUser = UserTestUtil.addUser();
+
+		GroupLocalServiceUtil.addUserGroupGroup(
```

Comment:
Hey @JorgeFerrer,

We need to manually create a site and associate it to test site membership inheritance from user group. These site groups have classNameId of `Group.class`. The group that's created automatically when creating a UserGroup represents the UserGroup entity and has a different classNameId of `UserGroup.class`.  Currently, the code does not take the UserGroup group into account, so we can't use the associated group for the test.